src/analysis/retail/priest/discipline/modules/spells/TwilightEquilibrium.tsx

```diff
@@ -0,0 +1,106 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import AtonementAnalyzer, { AtonementAnalyzerEvent } from '../core/AtonementAnalyzer';
+import { HOLY_DAMAGE_SPELLS, SHADOW_DAMAGE_SPELLS } from '../../constants';
+import { calculateEffectiveDamage, calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import { SpellLink } from 'interface';
+
+const TE_INCREASE = 0.15;
+
+class TwilightEquilibrium extends Analyzer {
+  healing = 0;
+  damage = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_PRIEST.TWILIGHT_EQUILIBRIUM_TALENT.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(AtonementAnalyzer.atonementEventFilter, this.onAtone);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+  }
+
+  onAtone(event: AtonementAnalyzerEvent) {
+    const { damageEvent } = event;
+    if (
+      !this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id) &&
+      !this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_SHADOW_BUFF.id)
+    ) {
+      return;
+    }
+
+    if (HOLY_DAMAGE_SPELLS.includes(damageEvent?.ability.guid ?? -1)) {
+      this.handleHolyHeal(event);
+    }
+
+    if (SHADOW_DAMAGE_SPELLS.includes(damageEvent?.ability.guid ?? -1)) {
+      this.handleShadowHeal(event);
+    }
+  }
+
+  handleHolyHeal(event: AtonementAnalyzerEvent) {
+    const { healEvent } = event;
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id)) {
+      this.healing += calculateEffectiveHealing(healEvent, TE_INCREASE);
+    }
+  }
+
+  handleShadowHeal(event: AtonementAnalyzerEvent) {
+    const { healEvent } = event;
+    if (this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_SHADOW_BUFF.id)) {
+      this.healing += calculateEffectiveHealing(healEvent, TE_INCREASE);
+    }
+  }
+
+  onDamage(event: DamageEvent) {
+    if (
+      !this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id) &&
+      !this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_SHADOW_BUFF.id)
+    ) {
+      return;
+    }
```

Comment:
What's the purpose of this early return, when all your other logic is wrapped in pretty much the same check?